databricks/koalas/base.py

```diff
@@ -299,10 +300,23 @@ def is_monotonic(self):
 
         >>> ser.rename("a").to_frame().set_index("a").index.is_monotonic
         True
+
+        >>> ser = ks.Series([5, 4, 3, 2, 1], index=[1, 2, 3, 4, 5])
+        >>> ser.is_monotonic
+        False
+
+        >>> ser.index.is_monotonic
+        True
         """
+        from databricks.koalas.frame import DataFrame
+        from databricks.koalas.series import _col
+        if self.hasnans:
+            return False
         col = self._scol
-        window = Window.orderBy(self._kdf._internal.index_scols).rowsBetween(-1, -1)
-        return self._with_new_scol((col >= F.lag(col, 1).over(window)) & col.isNotNull()).all()
+        sdf = self._internal.sdf.withColumn('id', monotonically_increasing_id())
+        window = Window.orderBy(sdf['id']).rowsBetween(-1, -1)
+        internal = _InternalFrame(sdf.select(col >= F.lag(col, 1).over(window)))
```

Comment:
This will invoke attaching default index which might cause performance penalty.
Shall we include the dummy id column and use it as index for the new DataFrame?

Reply:
Can we remove ` & col.isNotNull()`, btw?

Reply:
Thanks for reviewing!

i think it's not necessary because the handling of null value is already processed above as

```python
if self.hasnans:
    return False
```

Reply:
I see. hm, it runs a Spark job once, maybe we should avoid it?

Reply:
@ueshin yeah, i also think it should be better.

maybe we can avoid this by using spark dataframe only. i fixed it just before